### PR TITLE
Override for @material-ui/core

### PR DESCRIPTION
**Pull request by jollytoad**

### CODE REVIEW

File: overrides.json [hunk 1]
```diff
@@ -26,12 +26,12 @@
...
         "./index": null,
-        "./esm/*": null,
-        "./es/*": null,
         "./*": {
           "require": "./*/index.js",
           "import": "./esm/*/index.js"
-        }
+        },
+        "./esm/*": "./esm/*/index.js",
+        "./es/*": "./es/*/index.js"
```
Comment (guybedford) [inline overrides.json:34]: Do we really need both of these, can we not just expose eg `esm` which seems to be more popular?

File: overrides.json [hunk 2]
```diff
@@ -26,12 +26,12 @@
           "import": "./esm/index.js"
         },
         "./index": null,
-        "./esm/*": null,
-        "./es/*": null,
```
Comment (guybedford) [inline overrides.json]: Should we also null out `internals` and `umd`?

File: overrides.json (hunk 1 above)
Comment (jollytoad) [reply overrides.json:34]: tbh, I've no idea, `es/` is actually documented here... https://material-ui.com/guides/minimizing-bundle-size/#ecmascript ... so i guess there is the possibility that people use it.

Personally I wasn't aware of either of these before this, but i get the feeling the packages I should be resolving to `esm/` according to the rest of the doc ... but `esm/` support appears to be incomplete on jspm cdn atm.

Comment (guybedford) [reply overrides.json:34]: Right, I guess there's no harm in keeping it.

File: overrides.json [hunk 3]
```diff
@@ -16,5 +16,23 @@
...
+  },
+  "@material-ui/core": {
+    "4": {
+      "exports": {
+        ".": "./index.js",
+        "./index.js": {
+          "require": "./index.js",
+          "import": "./esm/index.js"
+        },
+        "./index": null,
```
Comment (guybedford) [inline overrides.json:28]:
```suggestion
        "./index": null,
        "./internal/*": null,
```

**Pull request merged**